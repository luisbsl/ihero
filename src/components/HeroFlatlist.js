import React from 'react'
import { FlatList } from 'react-native'

import HeroCard from './HeroCard'

export default class HeroFlatlist extends React.Component {
  render() {
    const { heroes, navigation } = this.props
    return (
      <FlatList
        style={{ marginTop: 10 }}
        data={heroes}
        keyExtractor={(item, index) => index.toString()}
        renderItem={
          ({ item }) =>
            <HeroCard
              hero={item}
              navigation={navigation} />
        }
      />
    )
  }
}