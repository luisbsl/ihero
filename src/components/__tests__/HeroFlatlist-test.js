import React from 'react'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import HeroFlatlist from '../HeroFlatlist'

test('renders correctly', () => {
  const heroes = [
    {
      "id": "1009187",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/60/5261a80a67e7d/standard_fantastic.jpg",
      "name": "Black Panther",
      "description": "",
      "comics": 543,
      "series": 175,
      "stories": 745
    },
    {
      "id": "1009220",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/3/50/537ba56d31087/standard_fantastic.jpg",
      "name": "Captain America",
      "description": "Vowing to serve his country any way he could, young Steve Rogers took the super soldier serum to become America's one-man army. Fighting for the red, white and blue for over 60 years, Captain America is the living, breathing symbol of freedom and liberty.",
      "comics": 2067,
      "series": 554,
      "stories": 3071
    },
    {
      "id": "1010338",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/80/5269608c1be7a/standard_fantastic.jpg",
      "name": "Captain Marvel (Carol Danvers)",
      "description": "",
      "comics": 421,
      "series": 140,
      "stories": 510
    },
    {
      "id": "1009282",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/5/f0/5261a85a501fe/standard_fantastic.jpg",
      "name": "Doctor Strange",
      "description": "",
      "comics": 672,
      "series": 213,
      "stories": 839
    },
    {
      "id": "1009368",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55/standard_fantastic.jpg",
      "name": "Iron Man",
      "description": "Wounded, captured and forced to build a weapon by his enemies, billionaire industrialist Tony Stark instead created an advanced suit of armor to save his life and escape captivity. Now with a new outlook on life, Tony uses his money and intelligence to make the world a safer, better place as Iron Man.",
      "comics": 2404,
      "series": 583,
      "stories": 3551
    },
    {
      "id": "1009583",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/7/20/527bb5d64599e/standard_fantastic.jpg",
      "name": "She-Hulk (Jennifer Walters)",
      "description": "Seriously wounded, young lawyer Jennifer Walters was given a blood transfusion by her cousin Bruce Banner. Usually in better control of her powers and temper than the Hulk, She-Hulk has been a high profile New York lawyer as well as an upstanding member of both the Avengers and Fantastic Four.",
      "comics": 405,
      "series": 131,
      "stories": 468
    },
    {
      "id": "1009664",
      "image": "http://i.annihil.us/u/prod/marvel/i/mg/d/d0/5269657a74350/standard_fantastic.jpg",
      "name": "Thor",
      "description": "As the Norse God of thunder and lightning, Thor wields one of the greatest weapons ever made, the enchanted hammer Mjolnir. While others have described Thor as an over-muscled, oafish imbecile, he's quite smart and compassionate.  He's self-assured, and he would never, ever stop fighting for a worthwhile cause.",
      "comics": 1582,
      "series": 433,
      "stories": 2419
    }
  ]

  const navigation = {
    navigate: {}
  }

  const tree = renderer.create(<HeroFlatlist heroes={heroes} navigation={navigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})