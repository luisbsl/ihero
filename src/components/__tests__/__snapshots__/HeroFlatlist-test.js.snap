// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "node": Object {
          "comics": 543,
          "id": "SGVybzoxMDA5MTg3",
          "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/60/5261a80a67e7d/standard_fantastic.jpg",
          "name": "Black Panther",
          "series": 175,
          "stories": 745,
        },
      },
      Object {
        "node": Object {
          "comics": 2067,
          "id": "SGVybzoxMDA5MjIw",
          "image": "http://i.annihil.us/u/prod/marvel/i/mg/3/50/537ba56d31087/standard_fantastic.jpg",
          "name": "Captain America",
          "series": 554,
          "stories": 3071,
        },
      },
      Object {
        "node": Object {
          "comics": 421,
          "id": "SGVybzoxMDEwMzM4",
          "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/80/5269608c1be7a/standard_fantastic.jpg",
          "name": "Captain Marvel (Carol Danvers)",
          "series": 140,
          "stories": 510,
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={undefined}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 10,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#e6e6e6",
                "borderColor": "#bfbfbf",
                "borderWidth": 2,
                "flexDirection": "row",
                "marginBottom": 10,
                "width": 740,
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "width": "30%",
                },
                undefined,
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/6/60/5261a80a67e7d/standard_fantastic.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 90,
                    "marginBottom": 10,
                    "marginTop": 10,
                    "width": 90,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 100,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": "80%",
                    "paddingTop": 25,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4d4d4d",
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                BLACK PANTHER
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginTop": 7,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  bgColor="#0066ff"
                  statName="Comics"
                  statNumber={543}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#0066ff",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        543
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Comics
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                />
                <View
                  bgColor="#00cc00"
                  statName="Series"
                  statNumber={175}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#00cc00",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        175
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Series
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#e6e6e6",
                "borderColor": "#bfbfbf",
                "borderWidth": 2,
                "flexDirection": "row",
                "marginBottom": 10,
                "width": 740,
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "width": "30%",
                },
                undefined,
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/3/50/537ba56d31087/standard_fantastic.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 90,
                    "marginBottom": 10,
                    "marginTop": 10,
                    "width": 90,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 100,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": "80%",
                    "paddingTop": 25,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4d4d4d",
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                CAPTAIN AMERICA
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginTop": 7,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  bgColor="#0066ff"
                  statName="Comics"
                  statNumber={2067}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#0066ff",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        2067
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Comics
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                />
                <View
                  bgColor="#00cc00"
                  statName="Series"
                  statNumber={554}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#00cc00",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        554
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Series
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#e6e6e6",
                "borderColor": "#bfbfbf",
                "borderWidth": 2,
                "flexDirection": "row",
                "marginBottom": 10,
                "width": 740,
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "width": "30%",
                },
                undefined,
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/6/80/5269608c1be7a/standard_fantastic.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 90,
                    "marginBottom": 10,
                    "marginTop": 10,
                    "width": 90,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 100,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": "80%",
                    "paddingTop": 25,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4d4d4d",
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                CAPTAIN MARVEL (CAROL DANVERS)
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "marginTop": 7,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  bgColor="#0066ff"
                  statName="Comics"
                  statNumber={421}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#0066ff",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        421
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Comics
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                />
                <View
                  bgColor="#00cc00"
                  statName="Series"
                  statNumber={140}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#00cc00",
                        "borderRadius": 2,
                        "height": 20,
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexBasis": 0,
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "flexShrink": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        140
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginLeft": 3,
                            "width": 90,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                            },
                            undefined,
                          ]
                        }
                      >
                        Series
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
